### PR TITLE
refact: adjust pom structure for directly compiling from root

### PR DESCRIPTION
**Pull request by heiyan-2020**
## Purpose of the PR

subtask of #2265 

Adjust the pom structure in order to directly build from root.

## Main Changes

- Remove some unused dependencies in `hugegraph-pd` module.
- Modify pom files for compiling and building.
- Manually compile the internal version of `hugegraph-core` and add it to the pom of store module as system dependencies.(temporal solution)
- The dependencies of packages imported through system dependencies will not be passed down, so we need to manually add some dependencies in `hugegraph-core` to the pom file of `hg-store-core`.

## Verifying these changes

- [x] Trivial rework / code cleanup without any test coverage. (No Need)
- [ ] Already covered by existing tests, such as *(please modify tests here)*.
- [ ] Need tests and can be verified as follows:

## Does this PR potentially affect the following parts?

- [x]  Nope
- [ ]  Dependencies (add/update license info) 
- [ ]  Modify configurations
- [ ]  The public API
- [ ]  Other affects

## Documentation Status

- [ ]  `Doc - TODO` 
- [ ]  `Doc - Done` 
- [x]  `Doc - No Need`

### CODE REVIEW

File: hugegraph-store/hg-store-dist/src/assembly/ext-lib/hugegraph-core-1.5.0.pom [hunk 1]
```diff
@@ -0,0 +1,441 @@
...
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+  License for the specific language governing permissions and limitations
+  under the License.
+  -->
+<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
```
Comment (VGalaxies) [inline hugegraph-store/hg-store-dist/src/assembly/ext-lib/hugegraph-core-1.5.0.pom:21]: In accordance with the hierarchical structure, the parent of `hugegraph-core` should be `hugegraph-server`. However, due to existing discrepancies between the community version of `hugegraph-server` and the internal version (such as `revision`), a temporary adjustment has been made for the purpose of building.

File: hugegraph-store/pom.xml [hunk 2]
```diff
@@ -49,6 +49,7 @@
         <maven.compiler.source>11</maven.compiler.source>
         <maven.compiler.target>11</maven.compiler.target>
         <log4j2.version>2.15.0</log4j2.version>
+        <top.level.dir>${project.basedir}/..</top.level.dir>
```
Comment (imbajin) [inline hugegraph-store/pom.xml:52]: WARN: ensure we don't have multi `top.level.dir` in project (if we have & need use it, add a comment for it)

File: hugegraph-store/hg-store-core/pom.xml [hunk 3]
```diff
@@ -121,9 +121,48 @@
             <version>${revision}</version>
             <scope>system</scope>
             <systemPath>
-                ${project.basedir}/../hg-store-dist/src/assembly/ext-lib/hugegraph-core-1.5.0.jar
+                ${top.level.dir}/hg-store-dist/src/assembly/ext-lib/hugegraph-core-1.5.0.jar
             </systemPath>
         </dependency>
+        <!-- tinkerpop -->
+        <dependency>
```
Comment (imbajin) [inline hugegraph-store/hg-store-core/pom.xml:128]: why we need add tinkerpop dependency here?

@VGalaxies any context?

File: hugegraph-store/hg-store-test/pom.xml [hunk 4]
```diff
@@ -210,7 +210,15 @@
             <version>${revision}</version>
             <scope>compile</scope>
         </dependency>
-
+        <dependency>
```
Comment (imbajin) [inline hugegraph-store/hg-store-test/pom.xml:213]:
```suggestion
        <!-- TODO: we need replace it with the apache version later -->
        <dependency>
```

File: hugegraph-store/hg-store-core/pom.xml [hunk 5]
```diff
@@ -121,9 +121,48 @@
             <version>${revision}</version>
             <scope>system</scope>
             <systemPath>
-                ${project.basedir}/../hg-store-dist/src/assembly/ext-lib/hugegraph-core-1.5.0.jar
+                ${top.level.dir}/hg-store-dist/src/assembly/ext-lib/hugegraph-core-1.5.0.jar
```
Comment (imbajin) [inline hugegraph-store/hg-store-core/pom.xml:124]: add a TODO comment for it  

and unify the param
image

File: hugegraph-store/hg-store-core/pom.xml (hunk 3 above)
Comment (heiyan-2020) [reply hugegraph-store/hg-store-core/pom.xml:128]: As mentioned in the last item of "Main change", tinkerpop is depended on by `hg-store-core`, but in internal version the dependency is imported through `hugegraph-core`. The current temporary solution is that importing `hugegraph-core` jar package through system dependency, whose dependencies cannot be passed on, so the dependencies need to be added again in `hg-store-core`

Comment (VGalaxies) [reply hugegraph-store/hg-store-core/pom.xml:128]: > why we need add tinkerpop dependency here?
> 
> @VGalaxies any context?

As stated by @heiyan-2020, assuming there are projects A (`hg-store-core`) and B (`hugegraph-core`), where project A depends on project B, and project B is a system dependency, the dependencies within project B will not be automatically inherited by project A.

File: hugegraph-store/pom.xml (hunk 2 above)
Comment (VGalaxies) [reply hugegraph-store/pom.xml:52]: Are you suggesting that we should only retain the `top.level.dir` property in the root pom? 

Additionally, provide all the occurrences of it:

image

File: hugegraph-store/hg-store-core/pom.xml (hunk 3 above)
Comment (imbajin) [reply hugegraph-store/hg-store-core/pom.xml:124]: > As mentioned in the last item of "Main change", tinkerpop is depended on by `hg-store-core`, and in internal version the dependency is passed on through `hugegraph-core` which imported as usal maven dependency. But the current temporary solution is that importing `hugegraph-core` jar package through system dependency. In this way, the dependencies in `hugegraph-core` cannot be passed on, so the dependencies need to be added again in `hg-store-core`

@heiyan-2020 get it, but we need add `comment/TODO` mark here? (maybe in next PR)

File: hugegraph-store/pom.xml (hunk 2 above)
Comment (imbajin) [reply hugegraph-store/pom.xml:53]: THX, just for warning

File: hugegraph-store/hg-store-dist/src/assembly/ext-lib/hugegraph-core-1.5.0.pom (hunk 1 above)
Comment (imbajin) [reply hugegraph-store/hg-store-dist/src/assembly/ext-lib/hugegraph-core-1.5.0.pom:21]: > In accordance with the hierarchical structure, the parent of `hugegraph-core` should be `hugegraph-server`. However, due to existing discrepancies between the community version of `hugegraph-server` and the internal version (such as `revision`), a temporary adjustment has been made for the purpose of building.

Update: miss the msg before, and we choose use a new version for all modules

**Pull request merged**